tools/const_finder/lib/const_finder.dart

```diff
@@ -32,9 +32,20 @@ class _ConstVisitor extends RecursiveVisitor<void> {
   final List<Map<String, dynamic>> constantInstances;
   final List<Map<String, dynamic>> nonConstantLocations;
 
+  // A cache of previously evaluated classes.
+  static Map<Class, bool> _classHeirarchyCache = <Class, bool>{};
   bool _matches(Class node) {
-    return node.enclosingLibrary.importUri.toString() == classLibraryUri &&
-      node.name == className;
+    if (node == null) {
```

Comment:
It looks like there's no need to test `node` for null, as `node.supers` doesn't include nulls.

Reply:
Cool, done.